File: Days/Days/Day1/day1.js
console.log('Finding if any two members of the array: 7, 10, 6, 8, 12, 3 are equal with k1: 17 or k2: 12');

let arr = [7, 10, 6, 8, 12, 3];
let k = 17;
let k2 = 12;
let found = false;
let found2 = false;

var valuesArr = [];

for (var i = 0; i < arr.length; i++) {
    if (valuesArr.find(e => e === k - arr[i])) {
        found = true;
    }

    if (valuesArr.find(e => e === k2 - arr[i])) {
        found2 = true;
    }

    valuesArr.push(arr[i]);
}

logFound(found, k);
logFound(found2, k2);

function logFound(found, no){ 
    if(found){
        console.log('Two such elements that add up to ' + no + ' were Found.');
    }
    else{
        console.log('Two such elements that add up to ' + no + ' were NOT Found.');
    }
}

function contains(a, obj) {
    for (var i = 0; i < a.length; i++) {
        if (a[i] === obj) {
            return true;
        }
    }
    return false;
}

File: Days/Days/Day1/Day1.cs
using System;
using System.Collections.Generic;

namespace Days.Day1
{
    public class Day1
    {
        public void TwoNumbersInArrayAddUp()
        {
            Console.WriteLine("Building the data-set");
            Console.Write("Input the No. of Elements in the array: ");
            int.TryParse(Console.ReadLine(), out int noOfElements);
            var arr = new List<int>();
            bool found = false;

            Console.WriteLine("Input elements: ");
            for(var i = 0; i < noOfElements; i++)
            {
                int.TryParse(Console.ReadLine(), out int elem);
                arr.Add(elem);
            }

            Console.WriteLine("Input sum: ");
            int.TryParse(Console.ReadLine(), out int k);

            Console.Write("Finding any two numbers in the list : ");
            foreach (var item in arr)
            {
                Console.Write(item + " ");
            }
            Console.WriteLine($"That add up to {k} ... ");

            HashSet<int> values = new HashSet<int>();
            for(int i = 0; i < arr.Count; i++)
            {                
                if (values.Contains(k - arr[i]))
                    found = true;

                values.Add(arr[i]);
            }

            if (found)
            {
                Console.WriteLine("Two such numbers were FOUND");
            }
            else
            {
                Console.WriteLine("Two such numbers were NOT FOUND");
            }            
        }
    }
}
